app/main/views/users.py

```diff
@@ -24,19 +24,19 @@ def find_user_by_email_address():
     if email_address:
         users = data_api_client.get_user(email_address=request.args.get("email_address"))
 
-    if users:
-        return render_template(
-            template,
-            users=[users['users']],
-            email_address=request.args.get("email_address")
-        )
-    else:
+    if not users and 'email_address' in request.args:
         flash('no_users', 'error')
         return render_template(
             template,
             users=list(),
             email_address=None
         ), 404
+    else:
+        return render_template(
+            template,
+            users=[users['users']] if users else list(),
+            email_address=request.args.get("email_address")
+        )
 
```

Comment:
It wouls be awesome if we could make this view like the other one. I much prefer that style of having a single return statement if we're using a single template because we always want to return the same things.

```
@main.route('/users', methods=['GET'])
@role_required('admin', 'admin-ccs-category')
def find_user_by_email_address():
    users = None
    email_address = None
    response_code = 200

    if "email_address" in request.args:
        email_address = request.args.get("email_address")
        if email_address:
            users = data_api_client.get_user(email_address=email_address)
        if not users:
            flash('no_users', 'error')
            response_code = 404

    return render_template(
        "view_users.html",
        users=[users['users']] if users else list(),
        email_address=email_address
    ), reponse_code
```

Feels like setting defaults for what we want to operate on at the top, keeping the logic in the middle, then demonstrating what we want to return really reduces clutter and makes the whole thing more readable. It breaks with the convention of failing/ dropping out early but easier supports DRY.

Totally happy with leaving it up to you two. This isn't a refactor PR 💯

Reply:
Nice one - yes we've done this now.